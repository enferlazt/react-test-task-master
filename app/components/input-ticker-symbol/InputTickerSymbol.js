import React, { useState } from 'react';
import { useHistory } from 'react-router';


const InputTickerSymbol = () => {
    const history = useHistory();
    const [inputField, setInputField] = useState('');

    function redirectTo() {
        if (inputField !== '') {
            history.push(`/${inputField}`);
        }
    }

    function enterHandler(e) {
        if (e.key === 'Enter') {
            redirectTo();
        }
    }

    return (
        <section className="ticker-input-block">
            <input className="ticker-input-block__input" value={inputField} onChange={(e) => setInputField(e.target.value)} onKeyDown={enterHandler} placeholder="Enter the valid ticker symbol"/>
            <button className="ticker-input-block__button" onClick={redirectTo}>Go To</button>
        </section>
    );
};

export default InputTickerSymbol;
