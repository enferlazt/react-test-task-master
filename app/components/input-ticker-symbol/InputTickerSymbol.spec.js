import { mount, shallow } from 'enzyme';
import React from 'react';
import InputTickerSymbol from './InputTickerSymbol';
import '../../../setupTests';

const setUp = () => shallow(<InputTickerSymbol/>);

describe("InputTickerSymbol component", () => {
    let component;

    beforeEach(() => {
        component = setUp();
    });

    it("render component", () => {
        expect(component).toMatchSnapshot();
    });

    it("ticker symbol search input", () => {
        const setInputFieldMock = jest.fn();
        const event = {
            target: { value: 'test' }
        };
        const component = mount(<InputTickerSymbol setInputField={setInputFieldMock} />);
        component.find('.ticker-input-block__input').simulate('change', event);
        expect(setInputFieldMock).toHaveBeenCalledWith('test');
    });
})