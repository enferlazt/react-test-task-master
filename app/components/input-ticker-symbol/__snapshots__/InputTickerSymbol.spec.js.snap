// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTickerSymbol component render component 1`] = `ShallowWrapper {}`;
